ProjectDplusDsSparse.py

```diff
@@ -39,13 +38,19 @@
     infilenames = [infilenames]
 enableSecPeak = inputCfg['enableSecPeak']
 isMC = inputCfg['isMC']
+isRedVar = inputCfg['isReducedVariables']
+isWithPtB = inputCfg['isWithPtB']
+shiftForRedVar = inputCfg['shiftForRedVariables']
```

Comment:
Maybe it would be convenient to have the pTB in a fixed position (i.e. right after pT) so that we can avoid this additional parameters. It is not super user friendly in case one has only an output with reduced variables, given that with the current implementation one should put the axis numbers of the non reduced version

Reply:
> it would be convenient to have the pTB in a fixed position (i.e. right after pT) so that we can avoid this additional parameters. 

You can't avoid having these additional parameters moving the pTB. You have to account for the difference between data and MC and the old and new MC "format".

> It is not super user friendly in case one has only an output with reduced variables, given that with the current implementation one should put the axis numbers of the non reduced version

The idea is that you define always the position w.r.t. to the full case and then "automatically" everything works also for the reduced case. You can do it the other way around but it's the same in the end.

Reply:
In case you do the other way around you have to shift by 1 only the MC axes and you don't have to care about the enumeration of the axes in case of complete sparses, but as I said it is minor and if you think it's more convenient like this, it's ok for me

Reply:
> In case you do the other way around you have to shift by 1 only the MC axes and you don't have to care about the enumeration of the axes in case of complete sparses

You still need to have variables for the "reduced" to "full" conversion otherwise you would need different cut files for the different cases

Reply:
Not if you put the pTB in a fixed position, right after the pT of the D (that was indeed my original suggestion). In this case you can just put the axis number of the data (whatever they are, either full or reduced) and tell the config to shift them by 1 in case of MC to account for the pTB which is right after the pT

Reply:
The position of the pTB is not related to the "full" vs "reduced" case in the end. What you suggest imply that you will have different `.yml` files depending on the sparse configuration. My understanding is that we should have only one `.yml` file, to support e.g. the possibility to have different sparse configurations for data and MC in `RunFullAnalysis.sh`

Reply:
I see your point now, indeed my solution would work if one has consistent configuration of sparses for data and MC, but has the advantage that one does not have to look how it is the enumeration of the axes in the full sparse if the reduced ones are used, so it is a matter of choice I would say (a bit more practical vs. a bit more flexible)

Reply:
> but has the advantage that one does not have to look how it is the enumeration of the axes in the full sparse if the reduced ones are used

But you will have to change the `.yml` depending on the configuration you use and this is more error-prone than defining the axis one time in the file for all the possible cases

Reply:
I promise that this is my last message :smile: but this is true only if you change the configuration of the sparse, and I don't see many reasons to do it, so IMO in the majority of the cases might be more error prone to put some axis number that does not correspond to your local output (but that's a personal view)

Reply:
> This is true only if you change the configuration of the sparse, and I don't see many reasons to do it
This is what I do for the Ds MC (which does not have size problems) to be able to do additional checks without running again on the GRID